src/main/java/coms/w4156/moviewishlist/controllers/WishlistController.java

```diff
@@ -26,78 +28,57 @@ public class WishlistController {
     @Autowired
     WishlistService wlService;
 
-    @GetMapping("/")
+    @Autowired
+    UserService userService;
+
+    @GetMapping
     public ResponseEntity<List<Wishlist>> getAllWishlists() {
         List<Wishlist> wlList = wlService.getAll();
-
-        if (wlList.isEmpty()) {
-            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
-        }
-
         return new ResponseEntity<>(wlList, HttpStatus.OK);
     }
 
     @GetMapping("/{id}")
-    public ResponseEntity<Wishlist> getWishlistById(@PathVariable String id) {
-        Long idLong = Long.parseLong(id, 10);
-
-        Optional<Wishlist> wl = wlService.findById(idLong);
-
-        if (!wl.isPresent()) {
-            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
-        } else {
-            Wishlist _wl = wl.get();
-            return new ResponseEntity<>(_wl, HttpStatus.OK);
-        }
+    public ResponseEntity<Wishlist> getWishlistById(@PathVariable long id) {
+        return wlService.findById(id)
+            .map(wishlist -> new ResponseEntity<>(wishlist, HttpStatus.OK))
+            .orElse(new ResponseEntity<>(HttpStatus.NO_CONTENT));
```

Comment:
cleaner, by using `.map` and `.orElse` methods of the `Optional` class.

Reply:
Much cleaner, nice